stats/series.go

```diff
@@ -0,0 +1,77 @@
+// Stats package includes simple statistics utilities
+package stats
+
+import (
+	"fmt"
+
+	"golang.org/x/exp/constraints"
+)
+
+type Number interface {
+	constraints.Integer | constraints.Float
+}
+
+// Series calculates cumulative statistics on a series of numbers
+// For streamlined use, all operations are designed dot be error-free, results
+// my not always be useful though.
+type Series[T Number] struct {
+	count uint
+	min   T
+	max   T
+	total T
+}
+
+func (s *Series[T]) Add(v T) {
+	if s.count <= 0 {
+		s.min = v
+		s.max = v
+		s.total = v
+	} else {
+		if s.min > v {
+			s.min = v
+		}
+		if s.max < v {
+			s.max = v
+		}
+		s.total += v
+	}
+	s.count += 1
+}
+
+func (s Series[T]) Len() uint { return s.count }
+func (s Series[T]) Min() T    { return s.min }
+func (s Series[T]) Max() T    { return s.max }
+func (s Series[T]) Total() T  { return s.total }
+
+func (s Series[T]) Avg() T {
+	if s.count <= 0 {
+		return 0
+	}
+	return s.total / T(s.count)
+}
+
+func (s Series[T]) Median() T {
```

Comment:
I think this should be midrange rather than median
```suggestion
func (s Series[T]) Midrange() T {
```

Reply:
will change to some other, more useful statistical property

Reply:
actually going to just drop it because I can't find any other useful statistic property that does not require me to store all the series data in order to calculate it.